src/lib/types.ts

```diff
@@ -88,6 +94,35 @@ export abstract class PostgrestBuilder<T> implements PromiseLike<PostgrestRespon
     return this
   }
 
+  retryingFetch(url: string, opts: any, retries: number, retryDelayMs: number): Promise<Response> {
+    // no automatic retries for non-read methods
+    if (!['GET', 'HEAD'].includes(this.method)) {
+      return Promise.resolve(this.fetch(url, opts))
+    }
```

Comment:
This is a good call. How should we handle PATCH requests that time out as on https://github.com/supabase/supabase/issues/6813?

Reply:
@darora Perhaps for this type of cases, we can increase the Cloudflare timeout?

https://support.cloudflare.com/hc/en-us/articles/115003011431-Troubleshooting-Cloudflare-5XX-errors#524error

> Enterprise customers can increase the 524 timeout up to 6000 seconds using the [proxy_read_timeout API endpoint](https://api.cloudflare.com/#zone-settings-change-proxy-read-timeout-setting).

Reply:
If PostgREST treats PATCH (and DELETE) as semantically idempotent it should be fine, but maybe safer to do it only for reads for now just to limit the impact, and note it as a TODO.

Reply:
> If PostgREST treats PATCH (and DELETE) as semantically idempotent it should be fin

@soedirgo They are idempotent if we stick to the definition in [rfc7231](https://www.rfc-editor.org/rfc/rfc7231#section-4.2.2):

> A request method is considered "idempotent" if the intended EFFECT ON THE SERVER of multiple identical requests with that method is the same as the effect for a single such request.
> It knows that repeating the request will have the same intended effect, even if the original request succeeded, though the response might differ.

The last paragraph applies(response might be different) when using `Prefer: return=representation`, so a DELETE would return contents the first time but an empty array the next time is tried on the same resource. Not sure if would be safe to retry in those cases, as it could cause application errors?

The way around that, would be not forcing the `Prefer: return=representation`(which IIRC, is also needed for RLS policies to work as intended without `return=minimal`) on `postgrest-js`, and then retry only for cases which do not specify that header.

Reply:
> retry only for cases which do not specify that header

Yup sounds good

> PATCH should be safe to retry on, even with Prefer: return=representation

Wouldn't it return different rows? I'm assuming return=representation returns the old rows

Reply:
I guess it still may not be idempotent even with `return=minimal`, because you might have a trigger that itself isn't idempotent (extreme example: on DELETE charge a Stripe payment w/ pg_net)

Reply:
> Wouldn't it return different rows? I'm assuming return=representation returns the old rows

Hm, `return=representation` returns the new updated row. Or do you mean the row would have been modified by another PATCH request?

Reply:
> return=representation returns the new updated row

Ah I see, you can ignore that part